group/assignment5/TestAlphabetFromDictionary.cpp

```diff
@@ -0,0 +1,161 @@
+    AlphabetFromDictionary alphabetFromDictionary(vector<string> ({"ART", "RAT", "CAC", "CAR"}));
+    EXPECT_FALSE(alphabetFromDictionary.isDictionaryConsistent());
+    vector<string> consistentSubDictionary = alphabetFromDictionary.makeDictionaryConsistent();
+    alphabetFromDictionary.setDictionary(consistentSubDictionary);
+    EXPECT_TRUE(alphabetFromDictionary.isDictionaryConsistent());
+    vector<char> alphabet = alphabetFromDictionary.getAlphabet();
+    set<char> actualChars(alphabet.begin(), alphabet.end());
+    set<char> expectedChars( {'A', 'R', 'C', 'T'});
+    unordered_map<char, int> charSequentialNumber;
+    EXPECT_EQ(4, alphabet.size());
+    for (int i = 0; i < alphabet.size(); i++) 
+        charSequentialNumber[alphabet[i]] = i;
+    EXPECT_EQ(expectedChars, actualChars);
+    EXPECT_TRUE(charSequentialNumber['A'] < charSequentialNumber['R']);
+    EXPECT_TRUE(charSequentialNumber['C'] < charSequentialNumber['R']);
+}
+
+void TestAlphabetFromDictionary::testInconsisntentDictionaryWithEmptyWords() {
+    AlphabetFromDictionary alphabetFromDictionary(vector<string> ({"", "ART", "RAT", "CAC", "", "CAR", ""}));
+    EXPECT_FALSE(alphabetFromDictionary.isDictionaryConsistent());
+    vector<string> consistentSubDictionary = alphabetFromDictionary.makeDictionaryConsistent();
+    alphabetFromDictionary.setDictionary(consistentSubDictionary);
+    EXPECT_TRUE(alphabetFromDictionary.isDictionaryConsistent());
+    vector<char> alphabet = alphabetFromDictionary.getAlphabet();
+    set<char> actualChars(alphabet.begin(), alphabet.end());
+    set<char> expectedChars( {'A', 'R', 'C', 'T'});
+    unordered_map<char, int> charSequentialNumber;
+    EXPECT_EQ(4, alphabet.size());
+    for (int i = 0; i < alphabet.size(); i++) 
+        charSequentialNumber[alphabet[i]] = i;
+    EXPECT_EQ(expectedChars, actualChars);
+    EXPECT_TRUE(charSequentialNumber['A'] < charSequentialNumber['R']);
+    EXPECT_TRUE(charSequentialNumber['C'] < charSequentialNumber['R']);
+}
+
+void TestAlphabetFromDictionary::testDictionaryWithComplexInconsistency() {
+    AlphabetFromDictionary alphabetFromDictionary(vector<string> ({"ART", "RAT", "CAC", "CAR", "RRC", "RRA"}));
+    EXPECT_FALSE(alphabetFromDictionary.isDictionaryConsistent());
+    vector<string> consistentSubDictionary = alphabetFromDictionary.makeDictionaryConsistent();
+    alphabetFromDictionary.setDictionary(consistentSubDictionary);
+    EXPECT_TRUE(alphabetFromDictionary.isDictionaryConsistent());
+    vector<char> alphabet = alphabetFromDictionary.getAlphabet();
+    set<char> actualChars(alphabet.begin(), alphabet.end());
+    set<char> expectedChars( {'A', 'R', 'C'});
+    unordered_map<char, int> charSequentialNumber;
+    EXPECT_EQ(3, alphabet.size());
+    for (int i = 0; i < alphabet.size(); i++) 
+        charSequentialNumber[alphabet[i]] = i;
+    EXPECT_EQ(expectedChars, actualChars);
+    EXPECT_TRUE(charSequentialNumber['A'] < charSequentialNumber['R']);
```

Comment:
This gives false and expected true

Reply:
As I wrote on Slack I think here should be:  EXPECT_TRUE(charSequentialNumber['C'] < charSequentialNumber['A']); in that case everything works well